------------------------------------------------------------------------

---
title: "Gene Set Enrichment Analysis"
author: "Lucy Chi"
date: "2023-03-22"
output: html_document
---

```{r}
# BiocManager::install("enrichplot")
# BiocManager::install("clusterProfiler")
# BiocManager::install("ggridges")
```

```{r}
library(tidyverse)
library(fgsea)
library(ggfortify)
library(msigdbr) 
library(data.table)
library(R.utils)
library(enrichplot)
library(DOSE)
library(clusterProfiler)
library(enrichplot)
library(biomaRt)
library(ggridges)
library(ReactomePA)
library(org.Hs.eg.db)
```

## Gene Sets for Downstream Gene Set Enrichment Analysis

### Obtain Molecular Signature Database (KEGG, reactome and GO pathways)

```{r}
knitr::kable(msigdbr::msigdbr_species())

kegg.human.db <- msigdbr::msigdbr(species = "human",
                                  category = "C2",
                                  subcategory = "CP:KEGG")
```

```{r}
#Access the REACTOME gene sets. 
#A standardized vocabulary of terms that describe gene function, cellular components, and biological processes.
reactome.human.db <- msigdbr(species = "human", 
                            category = "C2", 
                            subcategory = "CP:REACTOME")
```

```{r}
#Access the GO gene sets. 
#Using all GO gene sets: biological process, cellular component, and molecular function. 
GO.human.db <- msigdbr(species = "human", 
                            category = "C5")

```

### Obtain GWAS Catalog Information

```{r}
run.if.needed <- function(.file, .code) {
    if(!file.exists(.file)) { .code }
    stopifnot(file.exists(.file))
}
```

```{r}
gwas.tidy.file <- "gwas_catalog_tidy.tsv.gz"
run.if.needed(gwas.tidy.file, {

    gwas.file <- "gwas_catalog_v1.0-associations_e105_r2022-02-02.tsv.gz"

    run.if.needed(gwas.file, {
        url <- "https://www.ebi.ac.uk/gwas/api/search/downloads/full"
        .file <- str_remove(gwas.file, ".gz$")
        download.file(url, destfile = .file)
        gzip(.file)
        unlink(.file)
    })

    .dt <-
        fread(gwas.file, sep="\t", quote="") %>%
        dplyr::select(`MAPPED_GENE`, `DISEASE/TRAIT`, `PVALUE_MLOG`)

    .dt <- .dt[order(.dt$PVALUE_MLOG, decreasing = TRUE),
               head(.SD, 1),
               by = .(`MAPPED_GENE`, `DISEASE/TRAIT`)]

    .count <- .dt[, .(.N), by = .(`DISEASE/TRAIT`)]
    .dt <- left_join(.count[`N` >= 100, ], .dt)[nchar(`MAPPED_GENE`)> 0,]

    .dt <- .dt[,
               .(gene_symbol = unlist(strsplit(`MAPPED_GENE`, split="[ ,.-]+"))),
               by = .(`DISEASE/TRAIT`, PVALUE_MLOG)]
    .dt[, p.value := 10^(-PVALUE_MLOG)]

    fwrite(.dt, file=gwas.tidy.file)
})
```

### Genes associated with T2D disease

```{r}
#' Convert a number into a scientifically-formatted string
#' @param x number
num.sci <- function(x) {
    format(x, scientific=TRUE, digits = 2)
}

gwas.db <- fread(gwas.tidy.file)    
gwas.db[, gs_name := `DISEASE/TRAIT`]

t2d_genes<- gwas.db[str_detect(`DISEASE/TRAIT`, "[Dd]iabetes") & !is.na(gene_symbol)] %>%
    mutate(p.value = num.sci(p.value)) %>%
    dplyr::select(`gs_name`, `gene_symbol`, `p.value`, `PVALUE_MLOG`) 

## 3388 genes identified
```

### Load DE Gene List

```{r}
# read data frame for the DE genes
DEgenes<-readRDS("degT2d_DE.RDS")
```

### Data formatting for `fgsea` analysis

```{r}
# matched Ensembl IDs to probeIDs
DEgenes$affy_hugene_1_0_st_v1 <- row.names(DEgenes) 
head(DEgenes)

```

```{r}
# convert transcript cluster IDs (also desribed as probeIDs) into other gene ids
ensembl_human = useMart("ensembl",dataset="hsapiens_gene_ensembl")
ensemble_genes <- getBM(attributes=c("ensembl_gene_id", "affy_hugene_1_0_st_v1", "hgnc_symbol", "entrezgene_id"),
      filters = "affy_hugene_1_0_st_v1",
      values = rownames(DEgenes),
      useCache=FALSE,
      mart=ensembl_human)

```

```{r}
# data clean up and rename
ensemble_genes_clean <-
    ensemble_genes %>%  
    dplyr::rename(gene_symbol = hgnc_symbol) %>% # Will use human gene symbol
    as.data.table()

ensemble_genes_clean$affy_hugene_1_0_st_v1 <- as.character(ensemble_genes_clean$affy_hugene_1_0_st_v1)

## Add on the p-values & other information in DEgenes
ensembl_geneIDs <- ensemble_genes_clean %>% left_join(DEgenes, by = "affy_hugene_1_0_st_v1") 

saveRDS(ensembl_geneIDs, file = "ensembl_geneIDs.RDS")

# Show NA's & other stats
summary(ensemble_genes$entrezgene_id)
```

```{r}
#remove duplicated gene names
logFC_genes <- ensembl_geneIDs$logFC
# name the vector
names(logFC_genes) <- ensembl_geneIDs$ensembl_gene_id
# omit any NA values 
gene_list<-na.omit(logFC_genes)

# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)
```

```{r}
# make deg.ranks for gfsea analysis later

ensembl_geneIDs <- ensembl_geneIDs %>% mutate(log_adjP = -log10(adj.P.Val))

# deg.ranks <- ensembl_geneIDs %>%
#     arrange(P.Value) %>% 
#     as.data.table 
  
# deg.ranks <- deg.ranks %>% (function(.dt) { log_adjP <- .dt$log_adjP; names(log_adjP) <- .dt$gene_symbol; log_adjP })


# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
sorted_genes <-
    ensembl_geneIDs[order(ensembl_geneIDs$log_adjP, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(gene_symbol)]

# gene_names = ensembl_geneIDs[ensembl_geneIDs$ensembl_gene_id %in% dedup_ids$ENSEMBL,]
# Create a vector of the gene unuiverse
deg.ranks <- sorted_genes$log_adjP
# Name vector with ENTREZ ids
names(deg.ranks) <- sorted_genes$gene_symbol
# omit any NA values 
deg.ranks<-na.omit(deg.ranks)
# sort the list in decreasing order (required for clusterProfiler)
deg.ranks = sort(deg.ranks, decreasing = TRUE)


```

### Heat Map for T2D vs Healthy DE genes

```{r}
## TODO
```

## Rank-based Gene Set Enrichment Analysis

### Graphical Visualization

```{r}
ensembl_geneIDs %>% 
  arrange(adj.P.Val) %>% 
  ggplot(aes(1:nrow(ensembl_geneIDs), -log10(adj.P.Val))) +
    geom_point(stroke=0) +
    labs(x = "ranked genes", y = "-log(adj.P)") +
    geom_hline(yintercept = 1, lty = 2, colour="red") +
  geom_hline(yintercept = 1.5, lty = 2, colour="blue") +
    geom_hline(yintercept = 2, lty = 2, colour="green") +
    scale_y_continuous("negative log adjusted p-value",
                       breaks = c(1, 2, 4),
                       labels=function(x) num.sci(10^(-x)))

```

### KEGG Enrichment

```{r}
# for ppt presentation
kegg_enrich <- enrichKEGG(gene = ensembl_geneIDs$entrezgene_id,
                 organism     = 'human',
                 pvalueCutoff = 0.05)

# visualize the enriched pathway - the T2D pathway
browseKEGG(kegg_enrich, 'hsa04930')
```

```{r}
.gs <- c("KEGG_INSULIN_SIGNALING_PATHWAY",
         "KEGG_TYPE_II_DIABETES_MELLITUS",
         "KEGG_PANCREATIC_CANCER")

CUTOFF <- 1e-2 # q-value cutoff

.dt <-
    kegg.human.db %>%
    filter(`gs_name` %in% .gs) %>%
    left_join(ensembl_geneIDs) %>%    
    group_by(gs_name) %>%          # for each gene set
    arrange(adj.P.Val) %>%         # sort genes by p-value
    mutate(g = 1:n()) %>%          # add gene order index
    ungroup()


#' @param gene.dt gene-level statitsics with `gene_symbol` and `adj.P.Val`
#' @param geneset.dt gene set membership with `gene_symbol` and `gs_name`
run.hyper.test <- function(gene.dt, geneset.dt, cutoff = CUTOFF) {

    .genes <- unique(gene.dt[, .(gene_symbol, adj.P.Val)]) %>%
        na.omit()
    .sets <- as.data.table(geneset.dt)
    .sets <- .sets[gene_symbol %in% .genes$gene_symbol,
                   .(gene_symbol, gs_name)]

    .dt <- left_join(.sets, .genes, by = "gene_symbol") %>% 
        as.data.table()

    ## Total number of genes
    ntot <- length(unique(.dt$gene_symbol))
    ## Total number of significant DEGs
    nsig <- nrow(unique(.dt[adj.P.Val < cutoff, .(gene_symbol)]))
    ## Gene set size
    gs.size <- .dt[,
                   .(m = length(unique(gene_symbol))),
                   by = .(gs_name)]
    ## Gene set overlap size
    overlap.size <- .dt[adj.P.Val < cutoff,
                        .(q = length(unique(gene_symbol))),
                        by = .(gs_name)]

    left_join(gs.size, overlap.size, by = "gs_name") %>%
        mutate(`q` = if_else(is.na(`q`), 0, as.numeric(`q`))) %>% 
        mutate(n = `ntot` - `m`) %>%
        mutate(k = `nsig`) %>%
        mutate(p.val = phyper(`q`, `m`, `n`, `k`, lower.tail=FALSE)) %>%
        arrange(p.val) %>%
        as.data.table
}

deg_kegg <-
    ensembl_geneIDs[order(ensembl_geneIDs$adj.P.Val, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(affy_hugene_1_0_st_v1)]
deg_kegg<-na.omit(deg_kegg)

#genes significantly associated with the interaction effect from KEGG pathway
kegg_path <- run.hyper.test(deg_kegg,
                                kegg.human.db,
                                cutoff=1e-2)
kegg_path
knitr::kable(head(kegg_path, 10))

# (kegg_path, title="KEGG Pathway Enrichment", top_term=10)
```

### KEGG `fgsea` analysis

```{r}
# Make a function that can help us to prepare the gene list to feed into the fgsea arguments. 
make.gs.lol <- function(data) {
    data <- as.data.table(data) %>% unique()
    data_list <-
        data[, .(gene = .(gene_symbol)), by = .(gs_name)] %>%
        as.list()
    names <- data_list$gs_name
    return_value <- data_list$gene
    names(return_value) <- names
    return(return_value)
}
```

### KEGG Pathway Analysis

```{r}
#Convert the KEGG pathway tibble
kegg.lol <- kegg.human.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()
kegg.fgsea <- fgsea::fgsea(pathways = kegg.lol, stats = deg.ranks, scoreType = "pos")
```

### KEGG `fgsea` Analysis

```{r}
kegg.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]

kegg.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge)
```

```{r}
# filter significant pathways. Set p<0.05. 
kegg_significant_pathways <- kegg.fgsea[padj < 0.05]

# barplot(kegg_significant_pathways, showCategory=25, font.size = 7) 
```

There are 34 significant pathways.

### KEGG Pathway Visualization

```{r}
#some examples from KEGG pathway
plot1 <- plotEnrichment(kegg.lol[["KEGG_INSULIN_SIGNALING_PATHWAY"]],
             deg.ranks) 
plot2 <- plotEnrichment(kegg.lol[["KEGG_FATTY_ACID_METABOLISM"]],
              deg.ranks)
```

```{r}
geneids <-bitr(names(logFC_genes), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb="org.Hs.eg.db")

#delete duplicate IDs
dedup_ids = geneids[!duplicated(geneids[c("ENSEMBL")]),]
# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
geneIDs <-
    ensembl_geneIDs[order(ensembl_geneIDs$logFC, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(entrezgene_id)]

# gene_names = ensembl_geneIDs[ensembl_geneIDs$ensembl_gene_id %in% dedup_ids$ENSEMBL,]
# Create a vector of the gene unuiverse
kegg_gene_list <- geneIDs$logFC
# Name vector with ENTREZ ids
names(kegg_gene_list) <- geneIDs$entrezgene_id
# omit any NA values 
kegg_gene_list<-na.omit(kegg_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

#plotting
kk2 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = "hsa",
               minGSSize    = 3,
               maxGSSize    = 80,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
#enrichment map
emapplot(pairwise_termsim(kk2), cex.params = list(category_label = 0.5))

# category netplot
cnetplot(kk2, categorySize="pvalue", foldChange=gene_list, cex_label_category = 0.5) # this is prob not very useful

# Ridgeplot
ridgeplot(kk2) + labs(x = "enrichment distribution") + theme_ridges(font_size = 5)
## isoleucine degradation is down-regulated - a branched amino acid that will cause maple syrup urine disease when its concentration is too high

# https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/
```

### Reactome `fgsea` Analysis

```{r}
#Convert the REACTOME pathway tibble with the make.gs.lol function made previously. 
REACTOME.lol <- reactome.human.db %>% 
  dplyr :: select(gene_symbol, gs_name) %>% 
  make.gs.lol()
```

```{r}
# GSEA with fgsea()
reactome.fgsea <- fgsea::fgsea(pathways = REACTOME.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
#Show top 5 genes for each pathway:
# Here, we are "pasting" the first five genes in the leadingEdge column
reactome.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

```{r}
#Show result in table. 
reactome.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge)
```

```{r}
# filter significant pathways. Set p<0.05. 
reactome_significant_pathways <- reactome.fgsea[padj < 0.05]
reactome_significant_pathways
```

There are 68 significant pathways from REACTOME 'fgsea' analysis.

### Reactome Visualization

```{r}
reactome_gene_list <- geneIDs$logFC
# Name vector with ENTREZ ids
names(reactome_gene_list) <- geneIDs$entrezgene_id
# omit any NA values 
reactome_gene_list<-na.omit(reactome_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
reactome_gene_list = sort(reactome_gene_list, decreasing = TRUE)

```

```{r}
#plotting
reactome <- gsePathway(geneList = reactome_gene_list, 
                       organism = "human", 
               pvalueCutoff = 0.05,
               pAdjustMethod = "BH", 
              verbose=FALSE)
```

```{r}
#enrichment map
emapplot(pairwise_termsim(reactome), cex_label_category = 0.35, showCategory = 40)
```

```{r}
# category netplot
cnetplot(reactome, categorySize="pvalue", foldChange=gene_list, cex_label_category = 0.5, cex_label_gene = 0.5, node_label = "category")

# Ridgeplot
ridgeplot(reactome) + labs(x = "enrichment distribution") + theme_ridges(font_size = 5)

```

### GO `fgsea` Analysis

```{r}
#Convert the GO tibble with the make.gs.lol function made previously. 
GO.lol <- GO.human.db %>% 
  dplyr :: select(gene_symbol, gs_name) %>% 
  make.gs.lol()
```

```{r}
# GSEA with fgsea()
GO.fgsea <- fgsea::fgsea(pathways = GO.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
#Show top 5 genes for each pathway:
# Here, we are "pasting" the first five genes in the leadingEdge column
GO.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

```{r}
#Show result in table. 
GO.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% 
    knitr::kable()
```

```{r}
# filter significant pathways. Set p<0.05. 
GO_significant_pathways <- GO.fgsea[padj < 0.05]
GO_significant_pathways
```

There are 3358 significant pathways present in the GO gene sets.

### GO Analysis Visualization

```{r}
# take the log2 fold change 
go_gene_list <- geneIDs$logFC
# name the vector
names(go_gene_list) <- geneIDs$entrezgene_id
# omit any NA values 
go_gene_list<-na.omit(go_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
go_gene_list = sort(go_gene_list, decreasing = TRUE)
```

```{r}
# keytypes(org.Hs.eg.db)

go <- gseGO(geneList= go_gene_list,
             ont ="ALL",
             pvalueCutoff = 0.05,
            nPerm = 100000, 
             OrgDb = org.Hs.eg.db, 
             verbose = FALSE)
```

```{r}
# dotplot
# dotplot(gse, showCategory=10, split=".sign") + facet_grid(.~.sign)

# enrichment map 

# emapplot(pairwise_termsim(go), showCategory = 30, color = "p.adjust", cex_label_category = 0.35)


# https://yulab-smu.top/biomedical-knowledge-mining-book/clusterprofiler-go.html
```


```{r}
# Boxplot to be drawn here. 
```

### GWAS pathway Analysis

```{r}
gwas.lol <- gwas.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()
gwas.fgsea <- fgsea::fgsea(pathways = gwas.lol, stats = deg.ranks, scoreType = "pos")

gwas.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

### GWAS `fgsea` Analysis

```{r}
gwas.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% head()

gwas.fgsea
```

-   400 pathways are identified using GWAS pathway

### Disease Association with T2D based on the DE genes Using Genome wide annotation for Human Database

```{r}
FDR <- 0.05
filtered_genes <- ensembl_geneIDs %>% filter(P.Value < FDR)

filtered_genes <- filtered_genes[order(filtered_genes$P.Value),]
filtered_genes$entrezgene_id[1:30]

# de <- names(geneList)[abs(geneList) > 2]
unique(filtered_genes$entrezgene_id)

edo <- enrichDGN(unique(filtered_genes$entrezgene_id))

barplot(edo, showCategory=25, font.size = 7) 

# there seems to be a higher association with female diseases. T2D has a stronger influence on females?
```

## Test Over-representation

#### KEGG `goseq` analysis

Janet, I think we might not have time for this anymore
