---
title: "Gene Set Enrichment Analysis"
author: "Lucy"
date: "2023-03-22"
output: html_document
---

```{r}
# BiocManager::install("enrichplot")
# BiocManager::install("clusterProfiler")
```

```{r}
knitr::opts_chunk$set(echo = TRUE)
library(knitr)
library(tidyverse)
library(GEOquery)
library(fgsea)
library(ggfortify)
library(msigdbr) 
library(data.table)
library(R.utils)
library(enrichplot)
library(DOSE)
library(clusterProfiler)
library(enrichplot)
```

## Gene Sets for Downstream Gene Set Enrichment Analysis

### Obtain Molecular Signature Database (KEGG and reactome pathways)

```{r}
knitr::kable(msigdbr::msigdbr_species())

kegg.human.db <- msigdbr::msigdbr(species = "human",
                                  category = "C2",
                                  subcategory = "CP:KEGG")
```

```{r}
#Access the REACTOME gene sets. 
#A standardized vocabulary of terms that describe gene function, cellular components, and biological processes.
reactome.human.db <- msigdbr(species = "human", 
                            category = "C2", 
                            subcategory = "CP:REACTOME")
```

```{r}
#Access the GO gene sets. 
#Using all GO gene sets: biological process, cellular component, and molecular function. 
GO.human.db <- msigdbr(species = "human", 
                            category = "C5")

```
### Obtain GWAS Catalog Information

```{r}
run.if.needed <- function(.file, .code) {
    if(!file.exists(.file)) { .code }
    stopifnot(file.exists(.file))
}
```

```{r}
gwas.tidy.file <- "gwas_catalog_tidy.tsv.gz"
run.if.needed(gwas.tidy.file, {

    gwas.file <- "gwas_catalog_v1.0-associations_e105_r2022-02-02.tsv.gz"

    run.if.needed(gwas.file, {
        url <- "https://www.ebi.ac.uk/gwas/api/search/downloads/full"
        .file <- str_remove(gwas.file, ".gz$")
        download.file(url, destfile = .file)
        gzip(.file)
        unlink(.file)
    })

    .dt <-
        fread(gwas.file, sep="\t", quote="") %>%
        dplyr::select(`MAPPED_GENE`, `DISEASE/TRAIT`, `PVALUE_MLOG`)

    .dt <- .dt[order(.dt$PVALUE_MLOG, decreasing = TRUE),
               head(.SD, 1),
               by = .(`MAPPED_GENE`, `DISEASE/TRAIT`)]

    .count <- .dt[, .(.N), by = .(`DISEASE/TRAIT`)]
    .dt <- left_join(.count[`N` >= 100, ], .dt)[nchar(`MAPPED_GENE`)> 0,]

    .dt <- .dt[,
               .(gene_symbol = unlist(strsplit(`MAPPED_GENE`, split="[ ,.-]+"))),
               by = .(`DISEASE/TRAIT`, PVALUE_MLOG)]
    .dt[, p.value := 10^(-PVALUE_MLOG)]

    fwrite(.dt, file=gwas.tidy.file)
})
```

### Genes associated with T2D disease

```{r}
#' Convert a number into a scientifically-formatted string
#' @param x number
num.sci <- function(x) {
    format(x, scientific=TRUE, digits = 2)
}

gwas.db <- fread(gwas.tidy.file)    
gwas.db[, gs_name := `DISEASE/TRAIT`]

gwas.db[str_detect(`DISEASE/TRAIT`, "[Dd]iabetes") & !is.na(gene_symbol)] %>%
    head() %>%
    mutate(p.value = num.sci(p.value)) %>%
    dplyr::select(`gs_name`, `gene_symbol`, `p.value`, `PVALUE_MLOG`) %>%
    knitr::kable()
```

### Genes associated with obesity (\### might not need this \###)

```{r}
gwas.db[str_detect(`DISEASE/TRAIT`, "[Oo]besity") & !is.na(gene_symbol)] %>%
    head() %>%
    mutate(p.value = num.sci(p.value)) %>%
    dplyr::select(`gs_name`, `gene_symbol`, `p.value`, `PVALUE_MLOG`) %>%
    knitr::kable()
```

### Load DE Gene List

```{r}
DEgenes<-readRDS("degT2d_DE.RDS")
```

### Data formatting for fgsea analysis

```{r}
# read data frame for the DE genes
DEgenes
DEgenes$test = rownames(DEgenes)

# matched Ensembl IDs to probeIDs
DEgenes$affy_hugene_1_0_st_v1 <- row.names(DEgenes) 
DEgenes

```

```{r}
# convert transcript cluster IDs(also desribed as probeIDs) into Ensembl IDs

library(biomaRt)
ensembl_human = useMart("ensembl",dataset="hsapiens_gene_ensembl")
ensemble_genes <- getBM(attributes=c("ensembl_gene_id", "affy_hugene_1_0_st_v1", "hgnc_symbol", "entrezgene_id"),
      filters = "affy_hugene_1_0_st_v1",
      values = rownames(DEgenes),
      useCache=FALSE,
      mart=ensembl_human)

## TODO: removed the redundant Ensembl gene ID

ensemble_genes$affy_hugene_1_0_st_v1 <- as.character(ensemble_genes$affy_hugene_1_0_st_v1)

ensembl_geneIDs <- ensemble_genes %>% left_join(DEgenes, by = "affy_hugene_1_0_st_v1") 
# ensembl_geneIDs

saveRDS(ensembl_geneIDs, file = "ensembl_geneIDs.RDS")

# gene_id <- pull(ensembl_geneIDs, ensembl_gene_id)
# gene_values <- as.numeric(DEgenes$logFC)

sum(is.na(ensemble_genes$entrezgene_id))
summary(ensemble_genes$entrezgene_id)
```

### Heat Map for T2D vs Healthy DE genes

```{r}
## TODO
```

## Rank-based Gene Set Enrichment Analysis

### Graphical Visualization

```{r}
ggplot(ensembl_geneIDs, aes(1:nrow(ensembl_geneIDs), -log10(adj.P.Val))) +
    geom_point(stroke=0) +
    xlab("genes") +
    geom_hline(yintercept = 1, lty = 2, colour="red") +
  geom_hline(yintercept = 1.5, lty = 2, colour="blue") +
    geom_hline(yintercept = 2, lty = 2, colour="green") +
    scale_y_continuous("adjusted p-value",
                       breaks = c(1, 2, 4),
                       labels=function(x) num.sci(10^(-x)))

```

### KEGG Pathway Analysis

```{r}
# Make a function that can help us to prepare the gene list to feed into the fgsea arguments. 
make.gs.lol <- function(.dt) {
    .dt <- as.data.table(.dt) %>% unique()
    .list <-
        .dt[, .(gene = .(gene_symbol)), by = .(gs_name)] %>%
        as.list()
    .names <- .list$gs_name
    .ret <- .list$gene
    names(.ret) <- .names
    return(.ret)
}

#Convert the KEGG pathway tibble. 
kegg.lol <- kegg.human.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()

deg.ranks <- ensembl_geneIDs %>%
    arrange(P.Value) %>% 
    as.data.table 
  
deg.ranks<-  deg.ranks %>% mutate(log_adjP = -log10(adj.P.Val))

deg.ranks <- deg.ranks %>% (function(.dt) { log_adjP <- .dt$log_adjP; names(log_adjP) <- .dt$hgnc_symbol; log_adjP })

kegg.fgsea <- fgsea::fgsea(pathways = kegg.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
# enriched KEGG pathway
kegg_enrich <- enrichKEGG(gene = filtered_genes$entrezgene_id[1:30],
                 organism     = 'human',
                 pvalueCutoff = 0.05)

# visualize the enriched pathway
## Type II diabetes mellitus
browseKEGG(kegg_enrich, 'hsa04930')

```

### KEGG `fgsea` Analysis

```{r}
kegg.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]

kegg.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% head()
kegg.fgsea
```

```{r}
# filter significant pathways. Set p<0.05. 
kegg_significant_pathways <- kegg.fgsea[padj < 0.05]
kegg_significant_pathways

```

There are 57 significant pathways.

### Reactome `fgsea` Analysis

```{r}
#Convert the REACTOME pathway tibble with the make.gs.lol function made previously. 
REACTOME.lol <- reactome.human.db %>% 
  dplyr :: select(gene_symbol, gs_name) %>% 
  make.gs.lol()
```

```{r}
# GSEA with fgsea()
reactome.fgsea <- fgsea::fgsea(pathways = REACTOME.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
#Show top 5 genes for each pathway:
# Here, we are "pasting" the first five genes in the leadingEdge column
kegg.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

```{r}
#Show result in table. 
reactome.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% 
    knitr::kable()
```

```{r}
# filter significant pathways. Set p<0.05. 
reactome_significant_pathways <- reactome.fgsea[padj < 0.05]
reactome_significant_pathways
```

There are 85 significant pathways from REACTOME 'fgsea' analysis.

### GWAS pathway Analysis

```{r}
gwas.lol <- gwas.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()
gwas.fgsea <- fgsea::fgsea(pathways = gwas.lol, stats = deg.ranks, scoreType = "pos")

gwas.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

### GWAS `fgsea` Analysis

```{r}
gwas.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% head()

gwas.fgsea
```

-   400 pathways are identified using GWAS pathway

### Enrichment Score Plots for KEGG Pathways

```{r}
data$entrez = mapIds(kegg.human.db,
                    keys=row.names(data), #Column containing Ensembl gene ids
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals="first")
```

```{r}
#some examples from KEGG pathway

plot1 <- plotEnrichment(kegg.lol[["KEGG_INSULIN_SIGNALING_PATHWAY"]],
             deg.ranks) 
plot2 <- plotEnrichment(kegg.lol[["KEGG_FATTY_ACID_METABOLISM"]],
              deg.ranks)

library(gridExtra)
grid.arrange(plot1, plot2, ncol = 2, widths = c(1, 1), heights = c(1, 1))
#plotEnrichment(kegg.lol[["KEGG_FRUCTOSE_AND_MANNOSE_METABOLISM"]],
#               deg.ranks)
#plotEnrichment(kegg.lol[["KEGG_SNARE_INTERACTIONS_IN_VESICULAR_TRANSPORT"]],
#              deg.ranks)
#plotEnrichment(kegg.lol[["KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY"]],
#              deg.ranks)
#plotEnrichment(kegg.lol[["KEGG_WNT_SIGNALING_PATHWAY"]],
#             deg.ranks)

```

### Disease Association with T2D based on the DE genes Using Genome wide annotation for Human Databse

```{r}
filtered_genes <- ensembl_geneIDs %>% filter(P.Value < 0.01) 
filtered_genes <- filtered_genes[order(filtered_genes$P.Value),]
filtered_genes$entrezgene_id[1:30]

# de <- names(geneList)[abs(geneList) > 2]

edo <- enrichDGN(filtered_genes$entrezgene_id[1:30])

barplot(edo, showCategory=20, font.size = 8) 

```

```{r}
fgseaRes <- fgsea(pathways = kegg.lol, 
                  stats    = deg.ranks,
                  nperm=10000, maxSize=500,
                  scoreType = "pos")
head(fgseaRes[order(pval), ])
kegg_fgsea_pathway <- as.data.frame(kegg.fgsea)
kegg_topPathwaysUp <- fgseaRes[ES > 0][head(order(pval), n=10), kegg_fgsea_pathway$pathway]
kegg_topPathwaysDown <- fgseaRes[ES < 0][head(order(pval), n=10), kegg_fgsea_pathway$pathway]
kegg_topPathways <- c(kegg_topPathwaysUp, rev(kegg_topPathwaysDown))
plotGseaTable(kegg.lol[kegg_topPathways], deg.ranks, fgseaRes,
              gseaParam=0.5)

# collapsedPathways <- collapsePathways(fgseaRes_gwas[order(pval)][padj < 0.05], 
#                                       gwas.lol, deg.ranks)
# mainPathways <- fgseaRes[kegg_fgsea_pathway %in% collapsedPathways$mainPathways][
#                          order(-NES), kegg_fgsea_pathway]
# plotGseaTable(examplePathways[mainPathways], exampleRanks, fgseaRes, 
#               gseaParam = 0.5)
```
