------------------------------------------------------------------------

---
title: "Gene Set Enrichment Analysis"
author: "Lucy Chi"
date: "2023-03-22"
output: github_document
---

```{r}
# BiocManager::install("enrichplot")
# BiocManager::install("clusterProfiler")
# BiocManager::install("ggridges")
```

```{r}
library(tidyverse)
library(fgsea)
library(ggfortify)
library(msigdbr) 
library(data.table)
library(R.utils)
library(enrichplot)
library(DOSE)
library(clusterProfiler)
library(enrichplot)
library(ggridges)
library(ReactomePA)
library(org.Hs.eg.db)
library(biomaRt)
```

## Gene Sets for Downstream Gene Set Enrichment Analysis

### Obtain Molecular Signature Database (KEGG, reactome and GO pathways)

```{r}
knitr::kable(msigdbr::msigdbr_species())

kegg.human.db <- msigdbr::msigdbr(species = "human",
                                  category = "C2",
                                  subcategory = "CP:KEGG")
```

```{r}
#Access the REACTOME gene sets. 
#A standardized vocabulary of terms that describe gene function, cellular components, and biological processes.
reactome.human.db <- msigdbr(species = "human", 
                            category = "C2", 
                            subcategory = "CP:REACTOME")
```

```{r}
#Access the GO gene sets. 
#Using all GO gene sets: biological process, cellular component, and molecular function. 
GO.human.db <- msigdbr(species = "human", 
                            category = "C5")

```

### Obtain GWAS Catalog Information

```{r}
run.if.needed <- function(.file, .code) {
    if(!file.exists(.file)) { .code }
    stopifnot(file.exists(.file))
}
```

```{r}
gwas.tidy.file <- "gwas_catalog_tidy.tsv.gz"
run.if.needed(gwas.tidy.file, {

    gwas.file <- "gwas_catalog_v1.0-associations_e105_r2022-02-02.tsv.gz"

    run.if.needed(gwas.file, {
        url <- "https://www.ebi.ac.uk/gwas/api/search/downloads/full"
        .file <- str_remove(gwas.file, ".gz$")
        download.file(url, destfile = .file)
        gzip(.file)
        unlink(.file)
    })

    .dt <-
        fread(gwas.file, sep="\t", quote="") %>%
        dplyr::select(`MAPPED_GENE`, `DISEASE/TRAIT`, `PVALUE_MLOG`)

    .dt <- .dt[order(.dt$PVALUE_MLOG, decreasing = TRUE),
               head(.SD, 1),
               by = .(`MAPPED_GENE`, `DISEASE/TRAIT`)]

    .count <- .dt[, .(.N), by = .(`DISEASE/TRAIT`)]
    .dt <- left_join(.count[`N` >= 100, ], .dt)[nchar(`MAPPED_GENE`)> 0,]

    .dt <- .dt[,
               .(gene_symbol = unlist(strsplit(`MAPPED_GENE`, split="[ ,.-]+"))),
               by = .(`DISEASE/TRAIT`, PVALUE_MLOG)]
    .dt[, p.value := 10^(-PVALUE_MLOG)]

    fwrite(.dt, file=gwas.tidy.file)
})
```

### Genes associated with T2D disease

```{r}
#' Convert a number into a scientifically-formatted string
#' @param x number
num.sci <- function(x) {
    format(x, scientific=TRUE, digits = 2)
}

gwas.db <- fread(gwas.tidy.file)    
gwas.db[, gs_name := `DISEASE/TRAIT`]

t2d_genes<- gwas.db[str_detect(`DISEASE/TRAIT`, "[Dd]iabetes") & !is.na(gene_symbol)] %>%
    mutate(p.value = num.sci(p.value)) %>%
    dplyr::select(`gs_name`, `gene_symbol`, `p.value`, `PVALUE_MLOG`) 

t2d_genes <- t2d_genes %>% filter(t2d_genes$gs_name == "Type 2 diabetes")
## 2364 genes identified
```

### Load DE Gene List

```{r}
# read data frame for the DE genes
DEgenes<-readRDS("degT2d_DE.RDS")
```

### Data formatting for `fgsea` analysis

```{r}
# matched Ensembl IDs to probeIDs
DEgenes$affy_hugene_1_0_st_v1 <- row.names(DEgenes) 
head(DEgenes)

```

```{r}
# convert transcript cluster IDs (also desribed as probeIDs) into other gene ids
ensembl_human = useMart("ensembl",dataset="hsapiens_gene_ensembl")
ensemble_genes <- getBM(attributes=c("ensembl_gene_id", "affy_hugene_1_0_st_v1", "hgnc_symbol", "entrezgene_id"),
      filters = "affy_hugene_1_0_st_v1",
      values = rownames(DEgenes),
      useCache=FALSE,
      mart=ensembl_human)

```

```{r}
# data clean up and rename
ensemble_genes_clean <-
    ensemble_genes %>%  
    dplyr::rename(gene_symbol = hgnc_symbol) %>% 
    as.data.table()

ensemble_genes_clean$affy_hugene_1_0_st_v1 <- as.character(ensemble_genes_clean$affy_hugene_1_0_st_v1)

# add on the p-values & other information in DEgenes
ensembl_geneIDs <- ensemble_genes_clean %>% left_join(DEgenes, by = "affy_hugene_1_0_st_v1") 

saveRDS(ensembl_geneIDs, file = "ensembl_geneIDs.RDS")

# Show NA's & other stats
summary(ensemble_genes$entrezgene_id)
```

```{r}
#filter the list with logFC values for KEGG analysis later on
logFC_genes <- ensembl_geneIDs$logFC
# name the vector
names(logFC_genes) <- ensembl_geneIDs$ensembl_gene_id
# omit any NA values 
gene_list<-na.omit(logFC_genes)
# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)
```

```{r}
# Create a new dataframe sorted_genes which has only the filtered genes using log(adj.P)
ensembl_geneIDs <- ensembl_geneIDs %>% mutate(log_adjP = -log10(adj.P.Val))

sorted_genes <-
    ensembl_geneIDs[order(ensembl_geneIDs$log_adjP, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(gene_symbol)]

# Create a vector of the gene
deg.ranks <- sorted_genes$log_adjP
# Name vector with ENTREZ ids
names(deg.ranks) <- sorted_genes$gene_symbol
# omit any NA values 
deg.ranks<-na.omit(deg.ranks)
# sort the list in decreasing order (required for clusterProfiler)
deg.ranks = sort(deg.ranks, decreasing = TRUE)

```

## Rank-based Gene Set Enrichment Analysis

### KEGG Enrichment

```{r}
# # for ppt presentation
# kegg_enrich <- enrichKEGG(gene = ensembl_geneIDs$entrezgene_id,
#                  organism     = 'human',
#                  pvalueCutoff = 0.05)
# 
# # visualize the enriched pathway - the T2D pathway
# browseKEGG(kegg_enrich, 'hsa04930')
```

```{r}
# take the pathways that represent T2D
.kegg_info <- c("KEGG_INSULIN_SIGNALING_PATHWAY",
         "KEGG_TYPE_II_DIABETES_MELLITUS",
         "KEGG_PANCREATIC_CANCER")

# q-value cutoff
CUTOFF <- 1e-2 

.dt <-
    kegg.human.db %>%
    filter(`gs_name` %in% .kegg_info) %>%
    left_join(ensembl_geneIDs) %>%    
    group_by(gs_name) %>%          # for each gene set
    arrange(adj.P.Val) %>%         # sort genes by p-value
    mutate(g = 1:n()) %>%          # add gene order index
    ungroup()


#' @param gene.dt gene-level statitsics with `gene_symbol` and `adj.P.Val`
#' @param geneset.dt gene set membership with `gene_symbol` and `gs_name`
run.hyper.test <- function(gene.dt, geneset.dt, cutoff = CUTOFF) {

    .genes <- unique(gene.dt[, .(gene_symbol, adj.P.Val)]) %>%
        na.omit()
    .sets <- as.data.table(geneset.dt)
    .sets <- .sets[gene_symbol %in% .genes$gene_symbol,
                   .(gene_symbol, gs_name)]

    .dt <- left_join(.sets, .genes, by = "gene_symbol") %>% 
        as.data.table()

    ## Total number of genes
    ntot <- length(unique(.dt$gene_symbol))
    ## Total number of significant DEGs
    nsig <- nrow(unique(.dt[adj.P.Val < cutoff, .(gene_symbol)]))
    ## Gene set size
    gs.size <- .dt[,
                   .(m = length(unique(gene_symbol))),
                   by = .(gs_name)]
    ## Gene set overlap size
    overlap.size <- .dt[adj.P.Val < cutoff,
                        .(q = length(unique(gene_symbol))),
                        by = .(gs_name)]

    left_join(gs.size, overlap.size, by = "gs_name") %>%
        mutate(`q` = if_else(is.na(`q`), 0, as.numeric(`q`))) %>% 
        mutate(n = `ntot` - `m`) %>%
        mutate(k = `nsig`) %>%
        mutate(p.val = phyper(`q`, `m`, `n`, `k`, lower.tail=FALSE)) %>%
        arrange(p.val) %>%
        as.data.table
}

deg_kegg <-
    ensembl_geneIDs[order(ensembl_geneIDs$adj.P.Val, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(affy_hugene_1_0_st_v1)]
deg_kegg<-na.omit(deg_kegg)

#genes significantly associated with the interaction effect from KEGG pathway
kegg_path <- run.hyper.test(deg_kegg,
                                kegg.human.db,
                                cutoff=1e-2)
kegg_path
knitr::kable(head(kegg_path, 10))

# (kegg_path, title="KEGG Pathway Enrichment", top_term=10)
```

### KEGG `fgsea` analysis

```{r}
# Make a function that can help us to prepare the gene list to feed into the fgsea arguments. 
make.gs.lol <- function(data) {
    data <- as.data.table(data) %>% unique()
    data_list <-
        data[, .(gene = .(gene_symbol)), by = .(gs_name)] %>%
        as.list()
    names <- data_list$gs_name
    return_value <- data_list$gene
    names(return_value) <- names
    return(return_value)
}
```

### KEGG Pathway Analysis

```{r}
#Convert the KEGG pathway tibble
kegg.lol <- kegg.human.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()
kegg.fgsea <- fgsea::fgsea(pathways = kegg.lol, stats = deg.ranks, scoreType = "pos")
```

### KEGG `fgsea` Analysis

```{r}
kegg.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]

kegg.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge)
```

```{r}
# filter significant pathways. Set p<0.05. 
kegg_significant_pathways <- kegg.fgsea[padj < 0.05]
kegg_significant_pathways
```

There are 32 significant pathways.

### KEGG Pathway Visualization

```{r}
# get the top 20 pathways from kegg_significant_pathways based on the enrichment score. 
top20_kegg <- kegg_significant_pathways %>%
  arrange(desc(NES)) 
# %>%
#   head(20)
```

```{r}
# Plot the barplot. 
# Plot the 'size' column from the table onto x-axis, the the top 20 pathways on the y-axis
# Using the size because size represents the size of each pathway after removing genes not present in 'names(stats)'
kegg_plot<- ggplot(top20_kegg, aes(x = size, y = pathway, fill = padj)) +
  geom_bar(stat = "identity") +
  xlab("number of genes") +
  ylab("pathway") +  
  scale_fill_gradient(low = "gray70", high = "red") +
  ggtitle("KEGG GSEA result") +
  theme_minimal() +
  theme(axis.text.y = element_text(size = 4)) 
# scale_y_discrete(labels = lab)
kegg_plot
```

```{r}
# #some examples from KEGG pathway
# plot1 <- plotEnrichment(kegg.lol[["KEGG_INSULIN_SIGNALING_PATHWAY"]],
#              deg.ranks) 
# plot2 <- plotEnrichment(kegg.lol[["KEGG_FATTY_ACID_METABOLISM"]],
#               deg.ranks)
```

```{r}
geneids <-bitr(names(logFC_genes), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb="org.Hs.eg.db")

#delete duplicate IDs
dedup_ids = geneids[!duplicated(geneids[c("ENSEMBL")]),]
# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
geneIDs <-
    ensembl_geneIDs[order(ensembl_geneIDs$logFC, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(entrezgene_id)]

# gene_names = ensembl_geneIDs[ensembl_geneIDs$ensembl_gene_id %in% dedup_ids$ENSEMBL,]
# Create a vector of the gene unuiverse
kegg_gene_list <- geneIDs$logFC
# Name vector with ENTREZ ids
names(kegg_gene_list) <- geneIDs$entrezgene_id
# omit any NA values 
kegg_gene_list<-na.omit(kegg_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

#plotting
kk2 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = "hsa",
               minGSSize    = 3,
               maxGSSize    = 80,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
#enrichment map
emapplot(pairwise_termsim(kk2), cex.params = list(category_label = 0.5))


# Ridgeplot
ridgeplot(kk2) + labs(x = "enrichment distribution") + theme_ridges(font_size = 5)
## isoleucine degradation is down-regulated - a branched amino acid

# https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/
```

### Reactome `fgsea` Analysis

```{r}
#Convert the REACTOME pathway tibble with the make.gs.lol function made previously. 
REACTOME.lol <- reactome.human.db %>% 
  dplyr :: select(gene_symbol, gs_name) %>% 
  make.gs.lol()
```

```{r}
# GSEA with fgsea()
reactome.fgsea <- fgsea::fgsea(pathways = REACTOME.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
#Show top 5 genes for each pathway:
# Here, we are "pasting" the first five genes in the leadingEdge column
reactome.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

```{r}
#Show result in table. 
reactome.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge)
```

```{r}
# filter significant pathways. Set p<0.05. 
reactome_significant_pathways <- reactome.fgsea[padj < 0.05]
reactome_significant_pathways
```

There are 70 significant pathways from REACTOME 'fgsea' analysis.

### Reactome Visualization

```{r}
# get the top 20 pathways from reactome_significant_pathways based on the enrichment score. 
top20_reactome <- reactome_significant_pathways %>%
  arrange(desc(NES)) %>%
  head(20)
```

```{r}
lab = c('REACTOME_DEFECTIVE_CHSY1_CAUSES_TPBS', 'REACTOME_MET_RECEPTOR_ACTIVATION', 'REACTOME_DEFECTIVE_CHST14_CAUSES_ \n EDS_MUSCULOCONTRACTURAL_TYPE', 'REACTOME_SYNTHESIS_OF_PROSTAGLANDINS_PG_ \n AND_THROMBOXANES_TX', 'REACTOME_TP53_REGULATES_TRANSCRIPTION_OF_ \n DEATH_RECEPTORS_AND_LIGANDS', 'REACTOME_DERMATAN_SULFATE_BIOSYNTHESIS', 'REACTOME_DEFECTIVE_B4GALT7_CAUSES_EDS_ \n PROGEROID_TYPE', 'REACTOME_CHONDROITIN_SULFATE_BIOSYNTHESIS', 'REACTOME_INTERLEUKIN_7_SIGNALING', 'REACTOME_CONSTITUTIVE_SIGNALING_BY_ \n ABERRANT_P13K_IN_CANCER', 'REACTOME_DISEASES_ASSOCIATED_WITH_ \n GLYCOSAMINOGLYCAN_METABOLISM', 'REACTOME_EGFR_DOWNREGULATION', 'REACTOME_INTERLEUKIN_6_FAMILY_SIGNALING', 'REACTOME_SIGNALING_BY_ERBB2_IN_CANCER', 'REACTOME_CREATION_OF_C4_AND_C2_ACTIVATORS', 'REACTOME_RRNA_MODIFICATION_IN_THE_NUCLEUS_AND_CYTOSOL', 'REACTOME_INTERLEUKIN_20_FAMILY_SIGNALING', 'REACTOME_ROLE_OF_PHOSPHOLIPIDS_IN_PHAGOCYTOSIS', 'REACTOME_RAF_INDEPENDENT_MAPK1_3_ACTIVATION', 'REACTOME_DOWNSTREAM_SIGNALING_ \n OF_ACTIVATED_FGFR2')

# Plot the barplot. 
# Plot the 'size' column from the table onto x-axis, the the top 20 pathways on the y-axis
# Using the size because size represents the size of each pathway after removing genes not present in 'names(stats)'
reactome_plot<- ggplot(top20_reactome, aes(x = size, y = pathway, fill = padj)) +
  geom_bar(stat = "identity") +
  xlab("number of genes") +
  ylab("pathway") +  
  scale_fill_gradient(low = "gray70", high = "red") +
  ggtitle("REACTOME GSEA result") +
  theme_minimal() +
  theme(axis.text.y = element_text(size = 6)) + 
  scale_y_discrete(labels = lab)
reactome_plot
```

```{r}
reactome_gene_list <- geneIDs$logFC
# Name vector with ENTREZ ids
names(reactome_gene_list) <- geneIDs$entrezgene_id
# omit any NA values 
reactome_gene_list<-na.omit(reactome_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
reactome_gene_list = sort(reactome_gene_list, decreasing = TRUE)

```

```{r}
#plotting
reactome <- gsePathway(geneList = reactome_gene_list, 
                       organism = "human", 
               pvalueCutoff = 0.05,
               pAdjustMethod = "BH", 
              verbose=FALSE)
```

```{r}
#enrichment map
emapplot(pairwise_termsim(reactome), cex_label_category = 0.35, showCategory = 40)
```

```{r}
# Ridgeplot
ridgeplot(reactome) + labs(x = "enrichment distribution") + theme_ridges(font_size = 5)
```

### GO `fgsea` Analysis

```{r}
#Convert the GO tibble with the make.gs.lol function made previously. 
GO.lol <- GO.human.db %>% 
  dplyr :: select(gene_symbol, gs_name) %>% 
  make.gs.lol()
```

```{r}
# GSEA with fgsea()
GO.fgsea <- fgsea::fgsea(pathways = GO.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
#Show top 5 genes for each pathway:
# Here, we are "pasting" the first five genes in the leadingEdge column
GO.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

```{r}
#Show result in table. 
GO.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% 
    knitr::kable()
```

```{r}
# filter significant pathways. Set p<0.05. 
GO_significant_pathways <- GO.fgsea[padj < 0.05]
GO_significant_pathways
```

There are 1317 significant pathways present in the GO gene sets.

```{r}
# get the top 20 pathways from GO_significant_pathways based on the enrichment score. 
top20_GO <- GO_significant_pathways %>%
  arrange(desc(NES)) %>%
  head(20)
```

```{r}
# Names of the pathways are too long, so breaking them into two lines over here for grpah later. 
labs = c('GOBP_REGULATION_OF_RETINAL_GANGLION_CELL_ \n AXON_GUIDANCE', 'GOBP_NEGATIVE_REGULATION_OF_CELL_ \n PROLIFERATION_INVOLVED_IN_HEART_MORPHOGENESIS', 'HP_BURKITT_LYMPHOMA', 'GOBP_MONOCYTE_AGGREGATION', 'HP_ELEVATED_PROPORTION_OF_CD4_NEGATIVE_ \n CD8_NEGATIVE_ALPHA_BETA_REGULATORY_T_CELLS', 'GOMF_INTERLEUKIN_1_RECEPTOR_ACTIVITY', 'HP_INCREASED_B_CELL_COUNT', 'GOBP_RESPONSE_TO_CORTICOTROPIN_RELEASING_HORMONE', 'GOBP_POSITIVE_REGULATION_OF_GONAD_ \n DEVELOPMENT', 'GOBP_LEUKOTRIENE_SIGNALING_PATHWAY', 'GOBP_HYALURONAN_CATABOLIC_PROCESS', 'GOBP_ACTIVATION_OF_PROTEIN_KINASE_C_ACTIVITY', 'GOBP_REGULATION_OF_BRANCHING_INVOLVED_IN_ \n PROSTATE_GLAND_MORPHOGENESIS', 'GOBP_REGULATION_OF_MALE_GONAD_DEVELOPMENT', 'HP_ABNORMAL_PROPORTION_OF_DOUBLE_NEGATIVE_ \n ALPHA_BETA_REGULATORY_T_CELL', 'GOBP_NEGATIVE_REGULATION_OF_B_CELL_ \n DIFFERENTIATION', 'GOBP_DICHOTOMOUS_SUBDIVISION_OF_TERMINAL_ \n UNITS_INVOLVED_IN_SALIVARY_GLAND_BRANCHING', 'GOBP_PROSTATE_GLANDULAR_ACINUS_ \n MORPHOGENESIS', 'HP_ELLIPTOCYTOSIS', 'GOBP_RRNA_TRANSPORT')
      
# Plot the barplot. 
# Plot the 'size' column from the table onto x-axis, the the top 20 pathways on the y-axis
# Using the size because size represents the size of each pathway after removing genes not present in 'names(stats)'
go_barplot<-ggplot(top20_GO, aes(x = size, y = pathway, fill = padj)) +
  geom_bar(stat = "identity") +
  xlab("number of genes") +
  ylab("pathway") +  
  scale_fill_gradient(low = "gray70", high = "red") +
  ggtitle("GO GSEA result") +
  theme(axis.text.y = element_text(size = 6)) +
  scale_y_discrete(labels = labs)
go_barplot
```

### GO Analysis Visualization

```{r}
# take the log2 fold change 
# go_gene_list <- geneIDs$logFC
# name the vector
# names(go_gene_list) <- geneIDs$entrezgene_id
# omit any NA values 
# go_gene_list<-na.omit(go_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
# go_gene_list = sort(go_gene_list, decreasing = TRUE)
```

```{r}
# keytypes(org.Hs.eg.db)

# go <- gseGO(geneList= go_gene_list,
#             ont ="ALL",
#             pvalueCutoff = 0.05,
#            nPerm = 100000, 
#             OrgDb = org.Hs.eg.db, 
#             verbose = FALSE)
```

```{r}
# dotplot
# dotplot(gse, showCategory=10, split=".sign") + facet_grid(.~.sign)

# enrichment map 

# emapplot(pairwise_termsim(go), showCategory = 30, color = "p.adjust", cex_label_category = 0.35)


# https://yulab-smu.top/biomedical-knowledge-mining-book/clusterprofiler-go.html
```

### Disease Association with T2D based on the DE genes Using Genome wide annotation for Human Database

```{r}
FDR <- 0.05
filtered_genes <- ensembl_geneIDs %>% filter(P.Value < FDR)

filtered_genes <- filtered_genes[order(filtered_genes$P.Value),]
filtered_genes$entrezgene_id[1:30]

# de <- names(geneList)[abs(geneList) > 2]
unique(filtered_genes$entrezgene_id)

edo <- enrichDGN(unique(filtered_genes$entrezgene_id))

barplot(edo, showCategory=25, font.size = 7) 

# there seems to be a higher association with female diseases. T2D has a stronger influence on females?
```
