---
title: "GSE41762 Regression Analysis"
author: "Cindy Zhang + edit"
date: "2023-02-19"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(knitr)
library(tidyverse)
library(GEOquery)
library(limma)
library(ggfortify)
```

### Loading Data

```{r}
eset <- getGEO("GSE41762", getGPL = FALSE)[[1]]
# eset <- getGEO(filename = "~/Downloads/GSE41762_series_matrix.txt.gz", getGPL = FALSE)
head(eset, 10)
str(pData(eset))
head(pData(eset)[1:3,1:5]) %>% kable()
```

### Data Wrangling

```{r}
pData(eset) <- pData(eset) %>%
  mutate(sample_id = geo_accession) %>% 
  select("status:ch1",sample_id,"Sex:ch1", "bmi:ch1", "age:ch1") 
colnames(pData(eset)) <- c("statu","sample_id","sex","bmi","age")

pData(eset)$bmi %>% as.double()
pData(eset)$age %>% as.double()

pData(eset) <- pData(eset) %>% 
  mutate(status = case_when(
         grepl("Non-diabetic donor", statu) ~ "nont2d",
         grepl("Diabetic donor", statu) ~ "t2d")) %>%
  mutate(BMI = case_when(
    bmi >  30 ~ "over30",
    bmi < 30 ~ "below30"
  )) %>% 
  select(-c(bmi,statu))
```

### Arrange Factor Levels

```{r}
pData(eset) <- pData(eset) %>% 
  mutate(BMI = fct_relevel(BMI, "below30", "over30")) %>% 
  mutate(status = as.factor(status))
```

### match ID between matrices

```{r}
identical(colnames(exprs(eset)), pData(eset)$sample_id)
```

### Missing Values

-   The missing data are found in sample ID 49 through 77

```{r}
express <- exprs(eset) %>% 
  as.data.frame() 
#samples with missing values 
names(colSums(is.na(express))>0)
express <- na.omit(express)
```

### Combine Data

```{r}
# metaData
id <- colnames(express)
MetaData <- pData(eset) %>% 
  select(sample_id, status, BMI) %>% 
  mutate(samples = sample_id) %>% 
  filter(sample_id %in% id)

toLongerMeta <- function(expset) {
    stopifnot(class(expset) == "ExpressionSet")
    expressionMatrix <- longExpressionMatrix <- express %>% 
    as.data.frame() %>%
    rownames_to_column("gene") %>%
    pivot_longer(cols = !gene, 
                 values_to = "Expression",
                 names_to = "sample_id") %>%
    left_join(MetaData)
  return(expressionMatrix)
}

joint <- toLongerMeta(eset)
head(joint, 3) %>% kable()
identical(MetaData$sample_id, colnames(express))

# experiment_labels <- c(rep("Experiment 1 Samples", 48), rep("Experiment 2 Samples", 29))
# MetaData$experiment = experiment_labels
```

## PCA for Data Visualization

```{r}
express_scaled <- scale(t(express), center = TRUE, scale = TRUE)
s <- svd(express_scaled)

loadings <- s$v[, 1:3]

scores <- express_scaled %*% loadings

pca_data <- as_tibble(scores) %>% rename(PC1 = V1, PC2 = V2, PC3 = V3) 
pca_data$sample_id = colnames(express)

pca_data <- left_join(pca_data, MetaData, by = "sample_id")

ggplot(pca_data, aes(x=PC2, y=PC1, shape=status, color = BMI)) + geom_point(size=3)
ggplot(pca_data, aes(x=PC3, y=PC2, shape=status, color = BMI)) + geom_point(size=3)
ggplot(pca_data, aes(x=PC3, y=PC1, shape=status, color = BMI)) + geom_point(size=3)

#geom_text(aes(label = sample_id), size=2, nudge_y = 0.05)

# ggplot(pca_data, aes(x=PC1, y=PC2, color = experiment)) + geom_point(size=3)
# ggplot(pca_data, aes(x=PC2, y=PC3, color = experiment)) + geom_point(size=3)
# ggplot(pca_data, aes(x=PC3, y=PC1, color = experiment)) + geom_point(size=3)
```

-   Observe the sample distribution from T2D and BMI. Combine the data for the following study:

## Samples in Each Group

```{r, echo=FALSE}
table(MetaData$BMI, MetaData$status)
```

### Pivot Data Format

```{r}
toLonger <- function(expressionMatrix) {
    expressionMatrix <- longExpressionMatrix <- expressionMatrix %>% 
    as.data.frame() %>%
    rownames_to_column("gene") %>%
    pivot_longer(cols = !gene, 
                 values_to = "Expression",
                 names_to = "sample_id") 
  return(expressionMatrix)
}

options(repr.plot.width = 30, repr.plot.height =2)

toLonger(express) %>% 
  ggplot(aes(x=sample_id, y= Expression, color=sample_id)) +
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1),legend.position = "none") + 
  labs(x = "sample", y = "Gene Expression") 
```

## Linear Models

```{r}
modm <- model.matrix(~BMI*status, MetaData)
lmFitEb <- eBayes(lmFit(express, modm))
```

### DE genes in obese vs. nonobese healthy samples

```{r}
DEobH <- topTable(lmFitEb, number = Inf, adjust.method="BH", p.value = 0.05, coef= "BMIover30")

DEobH 
```

### DE genes in obese vs. nonobese T2D samples

```{r}
DEObT2d <- topTable(lmFitEb, number = Inf, adjust.method="BH", p.value = 0.05, coef= "BMIover30:statust2d") 
DEObT2d
```

### DE genes in T2D vs. Healthy in nonobese samples

```{r}
degT2dNonOb <- topTable(lmFitEb, number = Inf, adjust.method="BH", p.value = 0.05, coef= "statust2d")

degT2dNonOb
```

### Saving relevant data for aim 2 gene enrichment analysis

```{r}
saveRDS(degT2dNonOb, file = "ObvsNonObHealthy.RDS")
```

-   Note: Result can be loaded into Aim 2 analysis using the following code `readRDS("ObvsNonObHealthy.RDS")`

## Gene Sets for Downstream Gene Set Enrichment Analysis

### Obtain Molecular Signature Database - Species

```{r}
knitr::kable(msigdbr::msigdbr_species())

kegg.human.db <- msigdbr::msigdbr(species = "human",
                                  category = "C2",
                                  subcategory = "CP:KEGG")
```

### Obtain GWAS Catalog Information

```{r}
gwas.tidy.file <- "gwas_catalog_tidy.tsv.gz"
run.if.needed(gwas.tidy.file, {

    gwas.file <- "gwas_catalog_v1.0-associations_e105_r2022-02-02.tsv.gz"

    run.if.needed(gwas.file, {
        url <- "https://www.ebi.ac.uk/gwas/api/search/downloads/full"
        .file <- str_remove(gwas.file, ".gz$")
        download.file(url, destfile = .file)
        gzip(.file)
        unlink(.file)
    })

    .dt <-
        fread(gwas.file, sep="\t", quote="") %>%
        dplyr::select(`MAPPED_GENE`, `DISEASE/TRAIT`, `PVALUE_MLOG`)

    .dt <- .dt[order(.dt$PVALUE_MLOG, decreasing = TRUE),
               head(.SD, 1),
               by = .(`MAPPED_GENE`, `DISEASE/TRAIT`)]

    .count <- .dt[, .(.N), by = .(`DISEASE/TRAIT`)]
    .dt <- left_join(.count[`N` >= 100, ], .dt)[nchar(`MAPPED_GENE`)> 0,]

    .dt <- .dt[,
               .(gene_symbol = unlist(strsplit(`MAPPED_GENE`, split="[ ,.-]+"))),
               by = .(`DISEASE/TRAIT`, PVALUE_MLOG)]
    .dt[, p.value := 10^(-PVALUE_MLOG)]

    fwrite(.dt, file=gwas.tidy.file)
})
```

### Genes associated with T2D disease

```{r}
gwas.db <- fread(gwas.tidy.file)    
gwas.db[, gs_name := `DISEASE/TRAIT`]

gwas.db[str_detect(`DISEASE/TRAIT`, "[Dd]iabetes") & !is.na(gene_symbol)] %>%
    head() %>%
    mutate(p.value = num.sci(p.value)) %>%
    dplyr::select(`gs_name`, `gene_symbol`, `p.value`, `PVALUE_MLOG`) %>%
    knitr::kable()
```

### Genes associated with obesity  (\### might not need this \###)

```{r}
gwas.db[str_detect(`DISEASE/TRAIT`, "[Oo]besity") & !is.na(gene_symbol)] %>%
    head() %>%
    mutate(p.value = num.sci(p.value)) %>%
    dplyr::select(`gs_name`, `gene_symbol`, `p.value`, `PVALUE_MLOG`) %>%
    knitr::kable()
```

### Data formatting for fgsea analysis

```{r}
# read data frame for the DE genes
degT2dNonOb
degT2dNonOb$test = rownames(degT2dNonOb)

# matched Ensembl IDs to probeIDs
degT2dNonOb$affy_hugene_1_0_st_v1 <- row.names(degT2dNonOb) 
degT2dNonOb

# convert transcript cluster IDs(also desribed as probeIDs) into Ensembl IDs

library(biomaRt)
ensembl_human = useMart("ensembl",dataset="hsapiens_gene_ensembl")
ensemble_genes <- getBM(attributes=c("ensembl_gene_id", "affy_hugene_1_0_st_v1", "hgnc_symbol"),
      filters = "affy_hugene_1_0_st_v1",
      values = rownames(degT2dNonOb),
      useCache=FALSE,
      mart=ensembl_human)

# removed the redundant Ensembl gene ID
ensembl_geneIDs <- ensemble_genes[-9,]
ensembl_geneIDs$affy_hugene_1_0_st_v1 <- as.character(.ensembl_geneIDs$affy_hugene_1_0_st_v1)


ensembl_geneIDs <- ensembl_geneIDs %>% left_join(degT2dNonOb, by = "affy_hugene_1_0_st_v1") 
ensembl_geneIDs

saveRDS(ensembl_geneIDs, file = "ensembl_geneIDs.RDS")

```

## Rank-based Gene Set Enrichment Analysis

### Graphical Visualization

```{r}
ggplot(ensembl_geneIDs, aes(1:nrow(ensembl_geneIDs), -log10(adj.P.Val))) +
    geom_point(stroke=0) +
    xlab("genes") +
    geom_hline(yintercept = 1, lty = 2, colour="red") +
  geom_hline(yintercept = 1.5, lty = 2, colour="blue") +
    geom_hline(yintercept = 2, lty = 2, colour="green") +
    geom_hline(yintercept = 3, lty = 2, colour="white") +
    scale_y_continuous("adjusted p-value",
                       breaks = c(1, 2, 4),
                       labels=function(x) num.sci(10^(-x)))

# ggplot(ensembl_geneIDs, aes(1:nrow(ensembl_geneIDs), -log10(P.Value))) +
#     geom_point(stroke=0) +
#     xlab("genes") +
#     geom_hline(yintercept = 1, lty = 2, colour="red") +
#   geom_hline(yintercept = 1.5, lty = 2, colour="blue") +
#     geom_hline(yintercept = 2, lty = 2, colour="green") +
#     geom_hline(yintercept = 3, lty = 2, colour="white") +
#     scale_y_continuous("adjusted p-value",
#                        breaks = c(1, 2, 4),
#                        labels=function(x) num.sci(10^(-x)))

```

### KEGG Pathway Analysis

```{r}
make.gs.lol <- function(.dt) {
    .dt <- as.data.table(.dt) %>% unique()
    .list <-
        .dt[, .(gene = .(gene_symbol)), by = .(gs_name)] %>%
        as.list()
    .names <- .list$gs_name
    .ret <- .list$gene
    names(.ret) <- .names
    return(.ret)
}

kegg.lol <- kegg.human.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()

deg.ranks <- ensembl_geneIDs %>%
    arrange(P.Value) %>% 
    as.data.table 
  
deg.ranks<-  deg.ranks %>% mutate(log_adjP = -log10(adj.P.Val))

deg.ranks <- deg.ranks %>% (function(.dt) { log_adjP <- .dt$log_adjP; names(log_adjP) <- .dt$hgnc_symbol; log_adjP })

kegg.fgsea <- fgsea::fgsea(pathways = kegg.lol, stats = deg.ranks, scoreType = "pos")
```

### KEGG `fgsea` Analysis

```{r}
kegg.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]

kegg.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% head()
kegg.fgsea
```

-   7 pathways are identified that are involved in genetic mechanisms for T2D using KEGG pathway

### GWAS pathway Analysis

```{r}
gwas.lol <- gwas.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()
gwas.fgsea <- fgsea::fgsea(pathways = gwas.lol, stats = deg.ranks, scoreType = "pos")

gwas.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

### GWAS `fgsea` Analysis

```{r}
gwas.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% head()

gwas.fgsea
```

### Plotting (incomplete)

```{r}
plotEnrichment(kegg.lol[["KEGG_MAPK_SIGNALING_PATHWAY"]],
               deg.ranks)
plotEnrichment(kegg.lol[["KEGG_MATURITY_ONSET_DIABETES_OF_THE_YOUNG"]],
               deg.ranks)
plotEnrichment(kegg.lol[["KEGG_NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION"]],
               deg.ranks)
plotEnrichment(kegg.lol[["KEGG_SNARE_INTERACTIONS_IN_VESICULAR_TRANSPORT"]],
               deg.ranks)
plotEnrichment(kegg.lol[["KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY"]],
               deg.ranks)
plotEnrichment(kegg.lol[["KEGG_WNT_SIGNALING_PATHWAY"]],
               deg.ranks)

```

```{r}
fgseaRes_gwas <- fgsea(pathways = gwas.lol, 
                  stats    = deg.ranks,
                  nperm=10000, maxSize=500,
                  scoreType = "pos")
head(fgseaRes_gwas[order(pval), ])
gwas_fgsea_pathway <- as.data.frame(gwas.fgsea)
gwas_topPathwaysUp <- fgseaRes_gwas[ES > 0][head(order(pval), n=10), gwas_fgsea_pathway$pathway]
gwas_topPathwaysDown <- fgseaRes_gwas[ES < 0][head(order(pval), n=10), gwas_fgsea_pathway$pathway]
gwas_topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
# plotGseaTable(gwas.lol[topPathways], deg.ranks, fgseaRes,
#               gseaParam=0.5)


fgseaRes <- fgsea(pathways = kegg.lol, 
                  stats    = deg.ranks,
                  nperm=10000, maxSize=500,
                  scoreType = "pos")
head(fgseaRes[order(pval), ])
kegg_fgsea_pathway <- as.data.frame(kegg.fgsea)
kegg_topPathwaysUp <- fgseaRes[ES > 0][head(order(pval), n=10), kegg_fgsea_pathway$pathway]
kegg_topPathwaysDown <- fgseaRes[ES < 0][head(order(pval), n=10), kegg_fgsea_pathway$pathway]
kegg_topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
plotGseaTable(kegg.lol[topPathways], deg.ranks, fgseaRes,
              gseaParam=0.5)

# collapsedPathways <- collapsePathways(fgseaRes_gwas[order(pval)][padj < 0.05], 
#                                       gwas.lol, deg.ranks)
# mainPathways <- fgseaRes[kegg_fgsea_pathway %in% collapsedPathways$mainPathways][
#                          order(-NES), kegg_fgsea_pathway]
# plotGseaTable(examplePathways[mainPathways], exampleRanks, fgseaRes, 
#               gseaParam = 0.5)
```

```{r}
rnk.file <- system.file("extdata", "naive.vs.th1.rnk", package="fgsea")
gmt.file <- system.file("extdata", "human.reactome.gmt", package="fgsea")

ranks <- read.table(rnk.file,
                    header=TRUE, colClasses = c("character", "numeric"))
ranks <- setNames(ranks$t, ranks$ID)
str(ranks)

pathways <- gmtPathways(gmt.file)
str(head(pathways))

fgseaRes <- fgsea(pathways, ranks, minSize=15, maxSize=500)
head(fgseaRes)

# pathways <- reactomePathways(names(deg.ranks))
```
