---
title: "Gene-Set-Enrichment-Analysis"
author: "Lucy Chi & Janet Zhang" 
date: "2023-03-22"
output: github_document   
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# BiocManager::install("enrichplot")
# BiocManager::install("clusterProfiler")
# BiocManager::install("ggridges")


# Load all the required R packages. 
library(tidyverse)
library(fgsea)
library(ggfortify)
library(msigdbr) 
library(data.table)
library(R.utils)
library(enrichplot)
library(DOSE)
library(clusterProfiler)
library(ggridges)
library(ReactomePA)
library(org.Hs.eg.db)
library(biomaRt)
```

## Gene Sets for Downstream Gene Set Enrichment Analysis

### Obtain Molecular Signature Database (KEGG, reactome and C7 pathways)

```{r}
knitr::kable(msigdbr::msigdbr_species())
```

#### KEGG database

```{r}
#Access the KEGG gene sets. 
kegg.human.db <- msigdbr::msigdbr(species = "human",
                                  category = "C2",
                                  subcategory = "CP:KEGG")
```

#### reactome database

```{r}
#Access the REACTOME gene sets. 
#A standardized vocabulary of terms that describe gene function, cellular components, and biological processes.
reactome.human.db <- msigdbr(species = "human", 
                            category = "C2", 
                            subcategory = "CP:REACTOME")
```

#### PID database

```{r}
#Access the Pathway Interaction Database (PID) gene sets. This one specifically look at signaling pathways, molecular interactions, reaction kinetics, and cellular locations of signaling molecules involved in a pathway. 

pid.human.db <- msigdbr::msigdbr(species = "human",
                                  category = "C2",
                                  subcategory = "CP:PID")
```

#### ImmuneSigDB Database

```{r}
#Access the C7 Immunological gene sets. 
#Specifically looking at the ImmuneSigDB subcollection of C7, because this is related to a collection of cell types, states, and perturbations within the immune system collected from human and mouse. (note: The other C7 subcollection is specifically regarding vaccine response to 50 vaccines. We are not interested in vaccine responses in T2D, so not exploring this subcollection.)
IMMUNE.human.db <- msigdbr(species = "human", 
                           category = "C7", 
                           subcategory = "IMMUNESIGDB")

```

### Obtain GWAS Catalog Information

```{r}
run.if.needed <- function(.file, .code) {
    if(!file.exists(.file)) { .code }
    stopifnot(file.exists(.file))
}
```

```{r}
gwas.tidy.file <- "gwas_catalog_tidy.tsv.gz"

run.if.needed(gwas.tidy.file, {
    gwas.file <- "gwas_catalog_v1.0-associations_e105_r2022-02-02.tsv.gz"
    run.if.needed(gwas.file, {
        url <- "https://www.ebi.ac.uk/gwas/api/search/downloads/full"
        .file <- str_remove(gwas.file, ".gz$")
        download.file(url, destfile = .file)
        gzip(.file)
        unlink(.file)
    })
    .dt <-
        fread(gwas.file, sep="\t", quote="") %>%
        dplyr::select(`MAPPED_GENE`, `DISEASE/TRAIT`, `PVALUE_MLOG`)
    .dt <- .dt[order(.dt$PVALUE_MLOG, decreasing = TRUE),
               head(.SD, 1),
               by = .(`MAPPED_GENE`, `DISEASE/TRAIT`)]
    .count <- .dt[, .(.N), by = .(`DISEASE/TRAIT`)]
    .dt <- left_join(.count[`N` >= 100, ], .dt)[nchar(`MAPPED_GENE`)> 0,]
    .dt <- .dt[,
               .(gene_symbol = unlist(strsplit(`MAPPED_GENE`, split="[ ,.-]+"))),
               by = .(`DISEASE/TRAIT`, PVALUE_MLOG)]
    .dt[, p.value := 10^(-PVALUE_MLOG)]

    fwrite(.dt, file=gwas.tidy.file)
})
```

### Genes associated with T2D disease

```{r}
#' Convert a number into a scientifically-formatted string
#' @param x number
num.sci <- function(x) {
    format(x, scientific=TRUE, digits = 2)
}

gwas.db <- fread(gwas.tidy.file)    
gwas.db[, gs_name := `DISEASE/TRAIT`]

t2d_genes<- gwas.db[str_detect(`DISEASE/TRAIT`, "[Dd]iabetes") & !is.na(gene_symbol)] %>%
    mutate(p.value = num.sci(p.value)) %>%
    dplyr::select(`gs_name`, `gene_symbol`, `p.value`, `PVALUE_MLOG`) 
# removed t1d associated genes
t2d_genes <- t2d_genes %>% filter(t2d_genes$gs_name == "Type 2 diabetes")
#removed duplicates
t2d_genes <-
    t2d_genes[order(t2d_genes$p.value, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(gene_symbol)]
## 1693 genes identified

# export result as csv
write.csv(t2d_genes, "tsd.genes.csv", row.names=FALSE)
```

### Load DE Gene List

```{r}
# read data frame for the DE genes
DEgenes<-readRDS("degT2d_DEcopy.RDS")
```

### Data formatting for `fgsea` analysis

```{r}
# matched Ensembl IDs to probeIDs
DEgenes$affy_hugene_1_0_st_v1 <- row.names(DEgenes) 
head(DEgenes)

```

```{r}
# convert transcript cluster IDs (also desribed as probeIDs) into other gene ids
ensembl_human = useMart("ensembl",dataset="hsapiens_gene_ensembl")
ensemble_genes <- getBM(attributes=c("ensembl_gene_id", "affy_hugene_1_0_st_v1", "hgnc_symbol", "entrezgene_id"),
      filters = "affy_hugene_1_0_st_v1",
      values = rownames(DEgenes),
      useCache=FALSE,
      mart=ensembl_human)

```

```{r}
# data clean up and rename
ensemble_genes_clean <-
    ensemble_genes %>%  
    dplyr::rename(gene_symbol = hgnc_symbol) %>% 
    as.data.table()

ensemble_genes_clean$affy_hugene_1_0_st_v1 <- as.character(ensemble_genes_clean$affy_hugene_1_0_st_v1)

# add on the p-values & other information in DEgenes
ensembl_geneIDs <- ensemble_genes_clean %>% left_join(DEgenes, by = "affy_hugene_1_0_st_v1") 

# Show NA's & other stats
summary(ensemble_genes$entrezgene_id)
```

```{r}
#filter the list with logFC values for KEGG analysis later on
logFC_genes <- ensembl_geneIDs$logFC
# name the vector
names(logFC_genes) <- ensembl_geneIDs$ensembl_gene_id
# omit any NA values 
gene_list<-na.omit(logFC_genes)
# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)
```

```{r}
# Create a new dataframe sorted_genes which has only the filtered genes using log(adj.P)
ensembl_geneIDs <- ensembl_geneIDs %>% mutate(log_adjP = -log10(adj.P.Val))

sorted_genes <-
    ensembl_geneIDs[order(ensembl_geneIDs$log_adjP, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(gene_symbol)]

only_ensembl <- as.data.frame(sorted_genes$entrezgene_id)
write.csv(only_ensembl, "only_ensembl.csv", row.names=FALSE)

# Create a vector of the gene
deg.ranks <- sorted_genes$log_adjP
# Name vector with ENTREZ ids
names(deg.ranks) <- sorted_genes$gene_symbol
# omit any NA values 
deg.ranks<-na.omit(deg.ranks)
# sort the list in decreasing order (required for clusterProfiler)
deg.ranks = sort(deg.ranks, decreasing = TRUE)

```

## Rank-based Gene Set Enrichment Analysis

### `fgsea` analysis set-up

```{r}
# Make a function that can help us to prepare the gene list to feed into the fgsea arguments. 
make.gs.lol <- function(data) {
    data <- as.data.table(data) %>% unique()
    data_list <-
        data[, .(gene = .(gene_symbol)), by = .(gs_name)] %>%
        as.list()
    names <- data_list$gs_name
    return_value <- data_list$gene
    names(return_value) <- names
    return(return_value)
}
```

### KEGG `fgsea` Analysis

```{r}
# set seed.
set.seed(10)
#Convert the KEGG pathway tibble
kegg.lol <- kegg.human.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()
kegg.fgsea <- fgsea::fgsea(pathways = kegg.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
kegg.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]

kegg.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge)
```

```{r}
# filter significant pathways. Set p<0.05. 
kegg_significant_pathways <- kegg.fgsea[padj < 0.05]
kegg_significant_pathways

# export result as csv
fwrite(kegg_significant_pathways, file = "kegg_significant_pathways.csv")
```

There are 34 significant pathways.

### KEGG Pathway Visualization

```{r}
# arrange kegg_significant_pathways based on the p.adjust value. 
kegg_ordered <- kegg_significant_pathways %>%
  arrange(padj) 
```

```{r}
# Plot the barplot. 
# Plot the 'size' column from the table onto x-axis, the the top 20 pathways on the y-axis
# Using the size because size represents the size of each pathway after removing genes not present in 'names(stats)'
kegg_plot<- ggplot(kegg_ordered, aes(x = size, y = reorder(pathway, -pval), fill = padj)) +
  # note ordered by p.val over here b/c some of the pathways have the exact same padj values.
  geom_bar(stat = "identity") +
  xlab("number of genes") +
  ylab("pathway") +  
  scale_fill_gradient(low = "red", high = "grey70") +
  ggtitle("KEGG GSEA result") +
  theme_minimal() +
  theme(axis.text.y = element_text(size = 4)) 
kegg_plot
```

```{r}
geneids <-bitr(names(logFC_genes), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb="org.Hs.eg.db")

#delete duplicate IDs
dedup_ids = geneids[!duplicated(geneids[c("ENSEMBL")]),]
# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
geneIDs <-
    ensembl_geneIDs[order(ensembl_geneIDs$logFC, decreasing = TRUE),
                   head(.SD, 1),
                   by = .(entrezgene_id)]

# gene_names = ensembl_geneIDs[ensembl_geneIDs$ensembl_gene_id %in% dedup_ids$ENSEMBL,]
# Create a vector of the gene unuiverse
kegg_gene_list <- geneIDs$logFC
# Name vector with ENTREZ ids
names(kegg_gene_list) <- geneIDs$entrezgene_id
# omit any NA values 
kegg_gene_list<-na.omit(kegg_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

#plotting
kk2 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = "hsa",
               minGSSize    = 3,
               maxGSSize    = 80,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
#enrichment map
emapplot(pairwise_termsim(kk2), cex.params = list(category_label = 0.5))


# Ridgeplot
ridgeplot(kk2) + labs(x = "enrichment distribution") + theme_ridges(font_size = 5)
## isoleucine degradation is down-regulated - a branched amino acid

# https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/
```

### PID `fgsea` Analysis

```{r}
# set seed.
set.seed(10)

#Convert the PID pathway tibble
pid.lol <- pid.human.db %>% dplyr::select(gene_symbol, gs_name) %>% make.gs.lol()

#Run fgsea analysis. 
pid.fgsea <- fgsea::fgsea(pathways = pid.lol, stats = deg.ranks, scoreType = "pos")

```

```{r}
#Show top 5 genes for each pathway:
# Here, we are "pasting" the first five genes in the leadingEdge column
pid.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
#Show result in table.
pid.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge)
# filter significant pathways. Set p<0.05. 
pid_significant_pathways <- pid.fgsea[padj < 0.05]
pid_significant_pathways

# export result as csv
fwrite(pid_significant_pathways, "pid_significant_pathways.csv")
```

There are a total of 25 significant pathways in the PID gene sets.

### PID Visualization

```{r}
# arrange pid_significant_pathways based on the p.adjust value. 
pid_ordered <- pid_significant_pathways %>%
  arrange(padj) 

# Plot the barplot. 
# Plot the 'size' column from the table onto x-axis, the the top 20 pathways on the y-axis
# Using the size because size represents the size of each pathway after removing genes not present in 'names(stats)'
pid_plot<- ggplot(pid_ordered, aes(x = size, y = reorder(pathway, -pval), fill = padj)) +
  # note ordered by p.val over here b/c some of the pathways have the exact same padj values.
  geom_bar(stat = "identity") +
  xlab("number of genes") +
  ylab("pathway") +  
  scale_fill_gradient(low = "red", high = "grey70") +
  ggtitle("PID GSEA result") +
  theme_minimal() +
  theme(axis.text.y = element_text(size = 6)) 
pid_plot
```

### Reactome `fgsea` Analysis

```{r}
# set seed.
set.seed(10)
#Convert the REACTOME pathway tibble with the make.gs.lol function made previously. 
REACTOME.lol <- reactome.human.db %>% 
  dplyr :: select(gene_symbol, gs_name) %>% 
  make.gs.lol()
# GSEA with fgsea()
reactome.fgsea <- fgsea::fgsea(pathways = REACTOME.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
#Show top 5 genes for each pathway:
# Here, we are "pasting" the first five genes in the leadingEdge column
reactome.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

```{r}
#Show result in table. 
reactome.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge)
```

```{r}
# filter significant pathways. Set p<0.05. 
reactome_significant_pathways <- reactome.fgsea[padj < 0.05]
reactome_significant_pathways

# export result as csv
fwrite(reactome_significant_pathways, "reactome_significant_pathways.csv")
```

There are a total of 82 significant pathways from REACTOME 'fgsea' analysis.

### Reactome Visualization

```{r}
# get the top 20 pathways from reactome_significant_pathways based on the p.adjusted value. 
top20_reactome <- reactome_significant_pathways %>%
  arrange(padj) %>%
  head(20)
```

```{r}
#split the pathway names into two line instead of one line, because the names are too long on the y-axis. 
lab1 = c('REACTOME_MRNA_SPLICING', 'REACTOME_ROLE_OF_PHOSPHOLIPIDS_IN_PHAGOCYTOSIS', 'REACTOME_INTERLEUKIN_7_SIGNALING', 'REACTOME_PROCESSING_OF_CAPPED_INTRON_ \n CONTAINING_PRE_MRNA', 'REACTOME_GLYCOSAMINOGLYCAN_METABOLISM', 'REACTOME_INTRACELLULAR_SIGNALING_BY_SECOND_ \n MESSENGERS', 'REACTOME_RRNA_MODIFICATION_IN_THE_NUCLEUS_ \n AND_CYTOSOL', 'REACTOME_INTERLEUKIN_1_FAMILY_SIGNALING', 'REACTOME_SIGNALING_BY_RECEPTOR_TYROSINE_KINASES', 'REACTOME_INTERLEUKIN_4_AND_INTERLEUKIN_13_ \n SINALING', 'REACTOME_PI3K_AKT_SIGNALING_IN_CANCER', 'REACTOME_DISEASES_OF_SIGNAL_TRANSDUCTION_BY_ \n GROWTH_FACTOR_RECEPTORS_AND_SECOND_MESSENGERS', 'REACTOME_CONSTITUTIVE_SIGNALING_BY_ABERRANT_ \n PI3K_AKT_NETWORK', 'REACTOME_NEGATIVE_REGULATION_OF_THE_PI3K_ \n AKT_NETWORK', 'REACTOME_RRNA_PROCESSING', 'REACTOME_HEMOSTASIS', 'REACTOME_METABOLISM_OF_RNA', 'REACTOME_MAPK_FAMILY_SIGNALING_CASCADES', 'REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM', 'REACTOME_SIGNALING_BY_INTERLEUKINS') 

# Plot the barplot. 
# Plot the 'size' column from the table onto x-axis, the the top 20 pathways on the y-axis
# Using the size because size represents the size of each pathway after removing genes not present in 'names(stats)'
reactome_plot<- ggplot(top20_reactome, aes(x = size, y = reorder(pathway, -pval), fill = padj)) +
  # note ordered by p.val over here b/c some of the pathways have the exact same padj values.
  geom_bar(stat = "identity") +
  xlab("number of genes") +
  ylab("pathway") +  
  scale_fill_gradient(low = "red", high = "gray70") +
  ggtitle("REACTOME GSEA result") +
  theme_minimal() +
  theme(axis.text.y = element_text(size = 6))+
  scale_y_discrete(labels = lab1)
reactome_plot
```

### ImmuneSigDB 'fgsea' Analysis

```{r}
# set seed.
set.seed(10)
#Convert the Immune signature tibble with the make.gs.lol function made previously. 
immune.lol <- IMMUNE.human.db %>% 
  dplyr :: select(gene_symbol, gs_name) %>% 
  make.gs.lol()

#Gene set enrichment analysis with fgsea(). 
IMMUNE.fgsea <- fgsea::fgsea(pathways = immune.lol, stats = deg.ranks, scoreType = "pos")
```

```{r}
#Show top 5 genes for each pathway:
# Here, we are "pasting" the first five genes in the leadingEdge column
IMMUNE.fgsea[,
           topGenes := paste0(head(unlist(`leadingEdge`), 5), collapse=", "),
           by = .(pathway)]
```

```{r}
#Show result in table. 
IMMUNE.fgsea %>%
    arrange(pval) %>%
    head(10) %>% 
    dplyr::select(-leadingEdge) %>% 
    knitr::kable()
```

```{r}
# filter significant pathways. Set p<0.05. 
IMMUNE_significant_pathways <- IMMUNE.fgsea[padj < 0.05]
IMMUNE_significant_pathways

# export result as csv
fwrite(IMMUNE_significant_pathways, "IMMUNE_significant_pathways.csv")
```

There are a total of 1970 significant pathways in the ImmuneSigDB subset of the immunologic signature gene sets from MsigDB.

### ImmuneSigDB Visualization

```{r}
# get the top 20 pathways from GO_significant_pathways based on the p.adjust value. 
top20_IMMUNE <- IMMUNE_significant_pathways %>%
  arrange(padj) %>%
  head(20)
```

```{r}
# Names of the pathways are too long, so breaking them into two lines over here for grpah later. 
lab2 = c('GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_ \n MONOCYTES_UP', 
         'GSE2706_UNSTIM_VS_2H_LPS_AND_R848_DC_DN', 
         'GSE26343_UNSTIM_VS_LPS_STIM_MACROPHAGE_UP', 
         'GSE25085_FETAL_LIVER_VS_FETAL_BM_SP4_THYMIC_ \n IMPLANT_DN', 
         'GSE46606_IRF4MID_VS_WT_CD40L_IL2_IL5_DAY3_ \n STIMULATED_BCELL_UP', 
         'GSE30971_CTRL_VS_LPS_STIM_MACROPHAGE_WBP7_  \n HET_2H_UP', 
         'GSE9988_LOW_LPS_VS_CTRL_TREATED_MONOCYTE_UP', 
         'GSE28726_NAIVE_VS_ACTIVATED_NKTCELL_DN', 
         'GSE32986_CURDLAN_HIGHDOSE_VS_GMCSF_AND_ \n CURDLAN_HIGHDOSE_STIM_DC_DN', 
         'GSE14769_UNSTIM_VS_80MIN_LPS_BMDM_DN', 
         'GSE16385_ROSIGLITAZONE_IFNG_TNF_VS_IL4_STIM_ \n MACROPHAGE_UP', 
         'GSE6674_ANTI_IGM_VS_CPG_STIM_BCELL_DN',
         'GSE21670_UNTREATED_VS_IL6_TREATED_STAT3_KO_ \n CD4_TCELL_DN',
         'GSE9601_NFKB_INHIBITOR_VS_PI3K_INHIBITOR_ \n TREATED_HCMV_INF_MONOCYTE_DN', 
         'GSE27434_WT_VS_DNMT1_KO_TREG_DN',
         'GSE9988_ANTI_TREM1_VS_CTRL_TREATED_ \n MONOCYTES_UP',
         'GSE45739_NRAS_KO_VS_WT_ACD3_ACD28_STIM_ \n CD4_TCELL_UP',
         'GSE37605_FOXP3_FUSION_GFP_VS_IRES_GFP_TREG_ \n C57BL6_UP',
         'GSE21670_STAT3_KO_VS_WT_CD4_TCELL_DN',
         'GSE45365_NK_CELL_VS_CD11B_DC_DN')

# Plot the barplot. 
# Plot the 'size' column from the table onto x-axis, the the top 20 pathways on the y-axis
# Using the size because size represents the size of each pathway after removing genes not present in 'names(stats)'
IMMUNE_barplot<-ggplot(top20_IMMUNE, aes(x = size, y = reorder(pathway, -pval), fill = padj)) + 
  # note ordered by p.val over here b/c some of the pathways have the exact same padj values. 
  geom_bar(stat = "identity") +
  xlab("number of genes") +
  ylab("pathway") +  
  scale_fill_gradient(low = "red", high = "gray70") +
  ggtitle("IMMUNE GSEA result") +
  theme(axis.text.y = element_text(size = 6)) +
  scale_y_discrete(labels=lab2)
IMMUNE_barplot
```

### Disease Association with T2D based on the DE genes Using DisGeNET

```{r}
FDR <- 0.05
filtered_genes <- ensembl_geneIDs %>% filter(P.Value < FDR)

filtered_genes <- filtered_genes[order(filtered_genes$P.Value),]
filtered_genes$entrezgene_id[1:30]

edo <- enrichDGN(unique(filtered_genes$entrezgene_id))

barplot(edo, showCategory=25, font.size = 7) 

# there seems to be a higher association with female diseases. T2D may have a stronger influence on females
```
